### PR TITLE
Speeds up tome code

### PR DESCRIPTION
**Pull request by ghost**
Xhuis is making a bloodcult PR! He's abandoned clockcult! It's the end times! What will Joan do now that the only person who ever cared about his own gamemode abandoned it?!

Speeds up tome code by making things work with associative lists. Current tome code isn't much better than old tome code, and I figured with the knowledge imparted to me in my umbrage work could be put to some immediate good.

### CODE REVIEW

File: code/game/gamemodes/cult/ritual.dm [hunk 1]
```diff
@@ -12,6 +12,17 @@ This file contains the arcane tome files.
 	throw_speed = 2
 	throw_range = 5
 	w_class = WEIGHT_CLASS_SMALL
+	var/list/rune_types //Rune types that we can draw
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/ritual.dm]: why not make it static+check for LAZYLEN(rune_types), so that it only makes a list of runes once?

Comment (Xhuis) [inline code/game/gamemodes/cult/ritual.dm]: Ah, like a global variable that all tomes draw from? That works.

**Comment by GunHog** [comment]
Joan will follow you everywhere, Xhuis! You cannot escape!

**Comment by ShadowDeath6** [comment]
confirmed that joan will now take over maintaining both cults and the shadowling rework

**Comment by Xhuis** [comment]
I wouldn't mind having _someone_ attached to me, @GunHog 
@ShadowDeath6 I've kind of accepted that

File: code/game/gamemodes/cult/ritual.dm [hunk 2]
```diff
@@ -12,6 +12,17 @@ This file contains the arcane tome files.
 	throw_speed = 2
 	throw_range = 5
 	w_class = WEIGHT_CLASS_SMALL
+	var/list/keyword_runes //Rune types that need a keyword when drawing
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/ritual.dm]: you don't use this var

Comment (Xhuis) [inline code/game/gamemodes/cult/ritual.dm]: Good eye, this was left over from older code.

File: code/game/gamemodes/cult/runes.dm [hunk 3]
```diff
@@ -1,5 +1,6 @@
 /var/list/sacrificed = list() //a mixed list of minds and mobs
 var/list/non_revealed_runes = (subtypesof(/obj/effect/rune) - /obj/effect/rune/malformed)
+var/static/list/rune_types //Every rune that can be drawn by tomes
```
Comment (RemieRichards) [inline code/game/gamemodes/cult/runes.dm]: if it's on a type or in a proc, ```/static```
if it's not, ```/global```

File: code/game/gamemodes/cult/ritual.dm [hunk 4]
```diff
@@ -12,6 +12,17 @@ This file contains the arcane tome files.
...
+
+/obj/item/weapon/tome/New()
+	..()
+	if(!LAZYLEN(rune_types))
+		rune_types = list()
+		for(var/i_can_do_loops_now_thanks_remie in non_revealed_runes)
+			var/obj/effect/rune/R = i_can_do_loops_now_thanks_remie
```
Comment (XDTM) [inline code/game/gamemodes/cult/ritual.dm]: uh

File: code/game/gamemodes/cult/runes.dm (hunk 3 above)
Comment (Xhuis) [inline code/game/gamemodes/cult/runes.dm]: Good to know. The DM ref doc didn't include anything as far as I could tell, so I was left to guess.

File: code/game/gamemodes/cult/ritual.dm (hunk 4 above)
Comment (Xhuis) [inline code/game/gamemodes/cult/ritual.dm]: What? ;)

**Pull request merged**